mesa/space.py

```diff
@@ -856,7 +881,7 @@ def get_neighbors(self, node_id: int, include_center: bool = False) -> List[int]
 
     def move_agent(self, agent: Agent, node_id: int) -> None:
         """ Move an agent from its current node to a new node. """
-
+        assert isinstance(agent.pos, int)
```

Comment:
This assertion seems redundant if the user were to use mypy to make sure that `node_id` is `int`. If this assertion is added, then there are many places where similar assertions need to be added, and it might pollute the code.

Reply:
The assert is here to specifically make mypy happy. `agent.pos` might be of any type specified in the `agent.pos` type annotation and anything but int will throw an error in the `_remove_agent()` function (or, more specifically the call signature will not match)